### PR TITLE
Pixelmod

### PR DESCRIPTION
**Pull request by DaviesX**
Modified the IDecalRenderer interface.

### CODE REVIEW

File: pixelmod/src/main/java/org/e8yes/pixelmod/renderer/IDecalRenderer.kt [hunk 1]
```diff
@@ -8,5 +8,5 @@ import org.e8yes.pixelmod.repository.DecalHandle
  * possible.
  */
 interface IDecalRenderer {
-    fun render(src: Bitmap, decal: DecalHandle): Bitmap
+    fun render(src: Bitmap, detect: DecalHandle, mod: DecalHandle): Bitmap
```
Comment (zhaonian) [inline pixelmod/src/main/java/org/e8yes/pixelmod/renderer/IDecalRenderer.kt]: can we use something like `srcLogo` and `destLogo`?
or update the kdoc with @param detect, @param mod to explain what they are?

**Review by zhaonian — CHANGES_REQUESTED** [review]

File: pixelmod/src/main/java/org/e8yes/pixelmod/repository/ImageDecalRepository.kt [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+import android.graphics.Bitmap
+
+data class ImageDecal(val handle: DecalHandle, val path: String, val image: Bitmap)
+
 /** A collection of decals in the form of physical files. */
-class ImageDecalRepository(private val descriptor: DecalRepositoryDescriptor) : IDecalRepository {
+class ImageDecalRepository : IDecalRepository {
+
+    private val descriptor: DecalRepositoryDescriptor
+    private val decals: List<ImageDecal> = ArrayList()
```
Comment (jaredsburrows) [inline pixelmod/src/main/java/org/e8yes/pixelmod/repository/ImageDecalRepository.kt]: private val decals = ArrayList<ImageDecal>()

File: build.gradle [hunk 3]
```diff
@@ -9,7 +9,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.0'
+        classpath 'com.android.tools.build:gradle:3.5.1'
```
Comment (jaredsburrows) [inline build.gradle]: Why update this in this PR?

Comment (DaviesX) [reply build.gradle]: hmm, I don't know where this came from. I guess I pushed some update button in android studio, then it changed project build file on its own. I'll take it away.

File: pixelmod/src/main/java/org/e8yes/pixelmod/renderer/IDecalRenderer.kt (hunk 1 above)
Comment (DaviesX) [reply pixelmod/src/main/java/org/e8yes/pixelmod/renderer/IDecalRenderer.kt]: ok

**Review by zhaonian — APPROVED** [review]

**Pull request merged**